marker_rustc_driver/src/conversion/marker/common/span.rs

```diff
@@ -0,0 +1,70 @@
+use marker_api::{
+    ast::{ExpnInfo, FileInfo, FilePos, SpanPos, SpanSource},
+    prelude::Span,
+};
+
+use crate::conversion::marker::MarkerConverterInner;
+
+impl<'ast, 'tcx> MarkerConverterInner<'ast, 'tcx> {
+    pub fn to_span(&self, rustc_span: rustc_span::Span) -> Span<'ast> {
+        Span::new(
+            self.to_span_src_id(rustc_span.ctxt()),
+            // The driver resugars all expressions and spans, this should therefore
+            // only be true for spans from macro expansion.
+            rustc_span.from_expansion(),
+            self.to_span_pos(rustc_span.lo()),
+            self.to_span_pos(rustc_span.hi()),
+        )
+    }
+
+    pub fn to_span_pos(&self, byte_pos: rustc_span::BytePos) -> SpanPos {
+        SpanPos::new(byte_pos.0)
+    }
+
+    pub fn to_span_source(&self, rust_span: rustc_span::Span) -> SpanSource<'ast> {
+        let ctxt = rust_span.ctxt();
+        if ctxt.is_root() {
```

Comment:
With an early return the code should be one indentation level closer to the left side of the screen.

```rust
if !ctxt.is_root() {
    return SpanSource::Macro(self.alloc(self.to_expn_info(&ctxt.outer_expn_data())));
}

// The rest of the code here but without the indentation

```